lib/phoenix_token_auth/registration_flow.ex

```diff
@@ -0,0 +1,28 @@
+defmodule PhoenixTokenAuth.RegistrationFlow do
+  alias PhoenixTokenAuth.Mailer
+  alias PhoenixTokenAuth.Util
+
+  def user_created_by_username?(changeset) do
+    if changeset.valid? do
+      Util.repo.transaction(fn ->
+        case Util.repo.insert(changeset) do
```

Comment:
i think we don't need to check if the changeset is valid here. `repo.insert(...)` will do thatl also, as don't do anything besides inserting the record here, we don't need the transaction.